### ISSUE TITLE
Artisan generated resource controller vars don't match resource route params when model has multiple words.

### ISSUE THREAD

**Issue by curtisblackwell** [laravel/framework#18167]
- Laravel Version: 5.4.13

### Description:
When using Artisan to generate a model (and resource controller) with multiple words, the route param doesn't match the var names used in the controller.

The route param ends up snake case, while the var names are camel case.

### Steps To Reproduce:
```sh
artisan make:model -mcr ProductionOrder
```

```php
// routes/web.php
Route::resource('production-orders', 'ProductionOrderController');

// app/ProductionOrder.php (generated by Artisan)
public function show(ProductionOrder $productionOrder) {};

// wherever.php
dd(Route::getRoutes());
// #allRoutes: array:8 [
//   "HEADapi/user" => Illuminate\Routing\Route {#250}
//   "HEADproduction-orders" => Illuminate\Routing\Route {#240}
//   "HEADproduction-orders/create" => Illuminate\Routing\Route {#239}
//   "POSTproduction-orders" => Illuminate\Routing\Route {#238}
//   "HEADproduction-orders/{production_order}" => Illuminate\Routing\Route {#237}
//   "HEADproduction-orders/{production_order}/edit" => Illuminate\Routing\Route {#236}
//   "PATCHproduction-orders/{production_order}" => Illuminate\Routing\Route {#235}
//   "DELETEproduction-orders/{production_order}" => Illuminate\Routing\Route {#234}
// ]
```

**Comment by frdteknikelektro** [comment]
Yes i have this problem too, this issue is valid.

**Comment by stefanoruth** [comment]
Is there a problem with it not working or is it just the url formatting thats not fitting you? :)

**Comment by curtisblackwell** [comment]
URL formatting has nothing to do with it. It's the param being passed as `{production_order}` while the controller expects `productionController`.

I did learn a workaround:

```php
Route::resource('production-orders', 'ProductionOrderController', [
     'parameters' => ['production-orders' => 'productionOrder']
]);
```

However, this is still a :bug:.

**Comment by devcircus** [comment]
Why not 
```
Route::resource('productionOrders', 'ProductionOrderController');
```
and everything just works as it should

**Comment by stefanoruth** [comment]
if i remember correctly you should be able to name the variable anything you want you dont have to keep the snakeCase version

**Comment by frdteknikelektro** [comment]
@stefanoruth I think the same, but it's failing when you 'bind' it to the model.
just try it,
```
php artisan make:model ProductOrder -m -c -r
```
boom, it generate method like
```
public function show(ProductOrder $productOrder)
```
there where it goes failing the argument is just null value.
but if I change it to
```
publiic function show(ProductOrder $product_order)
```
It works. of course if I change it without bind it like
```
public function show($a)
```
, it also works, but $a is string not a model.

I think the problem is in artisan generation. (not only show method)

**Comment by frdteknikelektro** [comment]
@curtisblackwell Btw, i prefer underscore instead of dash. Seems like underscore is Laravel default. :smile_cat:

**Comment by curtisblackwell** [comment]
@devcircus …because it doesn't work as it should. See the first post.

<img width="678" alt="artisan_generated_resource_controller_vars_don_t_match_resource_route_params_when_model_has_multiple_words__ _issue__18167_ _laravel_framework" src="https://cloud.githubusercontent.com/assets/468029/23588456/5caf7cdc-0183-11e7-8513-3289d54deaea.png">

@stefanoruth Yes, you can rename it, but the point of Artisan is to make things more convenient. It doesn't just work when model names are multi-word in camel case.

As someone new to building Laravel apps, this is pretty confusing. It took me a long time to track down the problem, and even asking a more experienced Laravel dev was no help. Should be an easily-prevented problem. It's obviously a simple mistake made when writing the code. No way whomever wrote it intentionally set something to send snake case but receive camel case.

**Comment by devcircus** [comment]
I understand the issue, I guess I've just never thought of snake casing my resource routes. So for me, it's always worked perfectly, as I use:
```
Route::resource('productionOrders', 'ProductionOrderController');
```
Maybe dig into the source and submit a PR to make it work the way you use it.

**Comment by curtisblackwell** [comment]
The first argument is used for the slug, though. Odd to use camel case there.

I do plan to submit a PR at some point, but I don't have the time to figure everything out now. Given I'll likely encounter this on every app, I'm sure I won't forget.

**Comment by ametad** [comment]
Laravel magically will inject a initiated ProductOrder model with the id given in the url at the placeholder:
 {production_order} . 

https://laravel.com/docs/5.4/controllers#resource-controllers


```
Naming Resource Route Parameters
By default, Route::resource will create the route parameters for your resource routes based on the "singularized" version of the resource name. You can easily override this on a per resource basis by passing parameters in the options array. The parameters array should be an associative array of resource names and parameter names:

Route::resource('user', 'AdminUserController', ['parameters' => [
    'user' => 'admin_user'
]]);
The example above generates the following URIs for the resource's show route:

/user/{admin_user}

```

**Comment by LeonardChallis** [comment]
Has this been fixed? I was just looking at some loosely related bugs and tried to reproduce this but (with Laravel 5.5.2) it seems to work as expected. Can anyone else confirm?

**Comment by curtisblackwell** [comment]
I'm seeing no change. Resource route still passes snake-cased variable name, and Artisan-generated controller still expects camelcase.

**Comment by snapey** [comment]
Just tried this in response to a Laracasts thread.

Making model and controller as `php artisan make:model -mcr WorkRequest` creates controller methods expecting to be passed `$workRequest`

The fix appears to be to specify the route with snake case

 `Route::resource('work_requests', 'WorkRequestController');`

This causes the route to expect {work_request} which gets passed to the controller as `$workRequest` and matches the variables in the artisan generated resource controller

**Comment by istratos** [comment]
In addition to snapey's answer, you can also use dashes in the resource method and it'll work just as usual.

`Route::resource('my-route'), 'RouteController');`

Route will expect `{my_route}` and controller will expect `(MyRoute $myRoute)` but it'll work as usual. Seems like this is a new fix, since it didnt use to work.

**Comment by Hobadee** [comment]
A year later and this appears to still be an issue.

I'm new to Laravel, and this has been driving me crazy, as stuff was working fine for me until I used a compound word as a name.

The point of `make:model`, as I see it, is to create a working scaffold.  If it creates a broken scaffold that I have to fix, whats the point?  Also, as a n00b, I spent several hours trying to figure out what *I* was doing wrong before I realized the framework was the issue.  This issue really raises the bar for entry to Laravel, and it would be great to fix it.

```
$ php artisan -V
Laravel Framework 5.8.9
```
